activemodel/lib/active_model/type/string.rb

```diff
@@ -12,7 +12,12 @@ def changed_in_place?(raw_old_value, new_value)
       private
 
         def cast_value(value)
-          ::String.new(super)
+          case value
+          when ::String then ::String.new(value)
+          when true then "t".freeze
```

Comment:
It just sucks that we have to duplicate the conditionals and the true and false cases.